runtime/js-compute-runtime/builtins/request-response.cpp

```diff
@@ -2745,6 +2762,10 @@ JSObject *Response::create(JSContext *cx, JS::HandleObject response,
   JS::SetReservedSlot(response, static_cast<uint32_t>(Slots::Redirected), JS::FalseValue());
   JS::SetReservedSlot(response, static_cast<uint32_t>(Slots::IsUpstream),
                       JS::BooleanValue(is_upstream));
+  JS::SetReservedSlot(response, static_cast<uint32_t>(Slots::IsGripUpgrade),
+                      JS::BooleanValue(is_grip));
+  JS::SetReservedSlot(response, static_cast<uint32_t>(Slots::GripBackend),
+                      JS::PrivateValue(std::move(backend.release())));
```

Comment:
Could you move this assignment to right before the `return`? That way if the error return on line 2775 happens, the pointer will still get freed.